src/Services/Cda/CdaTemplateImportDispose.php

```diff
@@ -822,31 +825,35 @@ public function InsertImmunization($imm_array, $pid, CarecoordinationTable $care
                 $appTable->zQuery($qc_insert, array($value['cvx_code_text'], $value['cvx_code'], $ct_id));
             }
 
-            $q1_unit = "SELECT * FROM list_options WHERE list_id='drug_units' AND title=?";
-            $res_q1_unit = $appTable->zQuery($q1_unit, array($value['amount_administered_unit']));
-            foreach ($res_q1_unit as $val) {
-                $oid_unit = $val['option_id'];
-            }
-            if ($res_q1_unit->count() == 0) {
-                $lres = $appTable->zQuery("SELECT IFNULL(MAX(CONVERT(SUBSTRING_INDEX(option_id,'-',-1),UNSIGNED INTEGER))+1,1) AS option_id FROM list_options WHERE list_id = ?", array('drug_units'));
-                foreach ($lres as $lrow) {
-                    $oid_unit = $lrow['option_id'];
+            if (trim($value['amount_administered_unit'] ?? '') != '') {
```

Comment:
shouldn't have to compare against `''`? at least haven't need that for all of the php warnings fixed in the past...

Reply:
Don't know when it was introduced, but trim throws an error if its null.  I believe that's what you're talking about?  As far as the != '', I decided to do that instead of empty because if the value is 0 empty returns true.

Reply:
okay so we don't want `amount_administered_unit` values of `0` to be recorded, thanks

Reply:
ok, thought it was safer to err on the side of storing data, but if we want to skip saving a unit of 0, I'll switch it to use empty.